src/App.jsx

```diff
@@ -50,6 +60,8 @@ function App() {
 
   const [posts, setPosts] = useState([]);
   const [user, setUser] = useState(null);
+  const [open, setOpen] = useState(false);
+  const [curUser, setCurUser] = useState([]);
```

Comment:
is curUser really needed to shadow the user? I don't think we need a curUser when user and curUser both are same entity. About using to json, you may use it on the user variable or while passing it as prop to the profile route.

TLDR: We don't need unnecessary variables/useStates.

Reply:
Yes, you are right it's my fault. I create a new PR that uses the existing user state.
 
You can now review it.